For #1648: Correct send zold calculations.

paulodamaso opened the pull request:
For #1648:
- Corrected `send_zold.groovy` calculations
- Corrected test

0crat commented:
Job [#1656](https://github.com/zerocracy/farm/issues/1656) is now in scope, role is `REV`

<!-- https://www.0crat.com/footprint/C3NDPUA8L/2380458588, version: 0.30.9, hash: ${buildNumber} -->

0crat commented:
This pull request [#1656](https://github.com/zerocracy/farm/issues/1656) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2380731227); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

<!-- https://www.0crat.com/footprint/C3NDPUA8L/2380731240, version: 0.30.9, hash: ${buildNumber} -->

krzyk commented on src/main/groovy/com/zerocracy/stk/pm/cost/funding/send_zold.groovy:
```diff
@@ -41,34 +42,30 @@ def exec(Project project, XML xml) {
+  Vesting vesting = new Vesting(project).bootstrap()
+  Cash amount = new Cash.S(claim.param('amount'))
+  if (claim.hasParam('minutes') && vesting.exists(recipient)){
+    amount = amount.add(vesting.rate(recipient).mul(Integer.parseInt(claim.param('minutes'))) / 60)
```
@paulodamaso I think it shouldn't add to amount. If there is a vesting, it should use the vesting part, if there is no vesting it should use amount from claim. @g4s8 Am I correct here? Vesting rate is the whole rate of given person, so it should be used as zold amount (if there is vesting in project). Right?

g4s8 replied:
@paulodamaso @krzyk is right here - vesting rate includes cash rate

krzyk commented on src/main/groovy/com/zerocracy/stk/pm/cost/funding/send_zold.groovy:
```diff
@@ -41,34 +42,30 @@ def exec(Project project, XML xml) {
     'message',
     new Par('We just sent you %s ZLD through https://wts.zold.io')
-      .say(amount.decimal())
+      .say(amount)
```
@paulodamaso I think it should be still `.decimal`, because we are sending `123 ZLD`, not `$123 ZLD`

paulodamaso commented:
@krzyk I had removed `.decimal()` before because https://github.com/zerocracy/farm/issues/1598 , it was messing with the values I was using to test. Corrected the values and put `decimal` back, please take a look.

krzyk commented:
@paulodamaso thanks

krzyk commented:
@rultor merge

rultor commented:
> @rultor merge

@krzyk Thanks for your request. @g4s8 Please confirm this.

g4s8 commented:
@rultor merge

rultor commented:
> @rultor merge

@g4s8 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/15785-416292023)

rultor commented:
> @rultor merge

@g4s8 @paulodamaso Oops, I failed. You can see the full log [here](http://www.rultor.com/t/15785-416292023) (spent 16min)

```
	at com.sun.org.apache.xerces.internal.impl.XMLScanner.scanComment(XMLScanner.java:801)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanComment(XMLDocumentFragmentScannerImpl.java:1034)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:899)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:602)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:505)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:842)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:771)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:643)
	at net.sf.saxon.event.Sender.sendSAXSource(Sender.java:427)
	... 37 more

Running com.zerocracy.claims.BatchClaimsTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.45 sec - in com.zerocracy.claims.BatchClaimsTest
Running com.zerocracy.farm.ruled.RdItemTest
Exception in thread "Thread-650" java.lang.IllegalStateException: http://datum.zerocracy.com/latest/rules/pm/no-lost-boosts.xsl says: The boost for job "gh:test/test#143" exists in boosts.xml, but the job is absent in the WBS among 0 jobs.	at com.zerocracy.farm.ruled.RdRules.check(RdRules.java:130)	at org.cactoos.func.FuncOf.lambda$new$3(FuncOf.java:88)	at org.cactoos.func.FuncOf.apply(FuncOf.java:104)	at org.cactoos.scalar.And.lambda$null$0(And.java:132)	at org.cactoos.scalar.And.value(And.java:167)	at org.cactoos.scalar.And.value(And.java:60)	at org.cactoos.scalar.IoCheckedScalar.value(IoCheckedScalar.java:70)	at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:58)	at com.zerocracy.farm.ruled.RdRules.validate(RdRules.java:105)	at com.zerocracy.farm.ruled.RdItem.close(RdItem.java:137)	at com.zerocracy.pm.cost.Boosts.boost(Boosts.java:134)	at com.zerocracy.farm.ruled.RdItemTest.lambda$closesUnderlyingItemAnyway$1(RdItemTest.java:127)	at org.cactoos.func.FuncOf.lambda$new$3(FuncOf.java:88)	at org.cactoos.func.FuncOf.apply(FuncOf.java:104)	at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:58)	at org.cactoos.scalar.IoCheckedScalar.value(IoCheckedScalar.java:70)	at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:58)	at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:59)	at org.cactoos.func.RunnableOf.run(RunnableOf.java:89)	at java.lang.Thread.run(Thread.java:748)swallowed exception: java.lang.IllegalStateException('http://datum.zerocracy.com/latest/rules/pm/no-lost-boosts.xsl says: The boost for job "gh:test/test#55" exists in boosts.xml, but the job is absent in the WBS among 0 jobs.')
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.269 sec - in com.zerocracy.farm.ruled.RdItemTest
Running com.zerocracy.claims.ClaimSignatureTest
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.763 sec - in com.zerocracy.claims.ClaimSignatureTest
Running com.zerocracy.tk.project.TkClaimTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 76.09 sec - in com.zerocracy.tk.project.TkClaimTest
Running com.zerocracy.pmo.BlanksTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.241 sec - in com.zerocracy.pmo.BlanksTest
Running com.zerocracy.claims.proc.BrigadeProcTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.321 sec - in com.zerocracy.claims.proc.BrigadeProcTest
Running com.zerocracy.farm.reactive.AsyncFlushTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.371 sec - in com.zerocracy.farm.reactive.AsyncFlushTest
Running com.zerocracy.pmo.PeopleTest
Tests run: 42, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.009 sec - in com.zerocracy.pmo.PeopleTest
Running com.zerocracy.tools.SignatureTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0 sec - in com.zerocracy.tools.SignatureTest

Results :

Tests in error: 
  BundlesTest.oneBundleWorksFine:264->lambda$oneBundleWorksFine$4:243 » IllegalArgument

Tests run: 548, Failures: 0, Errors: 1, Skipped: 12

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 12:55 min
[INFO] Finished at: 2018-08-27T17:09:35+00:00
[INFO] Final Memory: 94M/595M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project farm: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/r/repo/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
container 174cdc886288c2bd94171de7963682bcac2997f75eb4e688647854a43d17f722 is dead
Mon Aug 27 19:10:00 CEST 2018
```

g4s8 commented:
@rultor merge

rultor commented:
> @rultor merge

@g4s8 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/15785-416324668)

rultor commented:
> @rultor merge

@g4s8 Done! FYI, the full log is [here](http://www.rultor.com/t/15785-416324668) (took me 21min)

0crat commented:
Job was finished in 35 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

<!-- https://www.0crat.com/footprint/C3NDPUA8L/2391940068, version: 0.30.9, hash: ${buildNumber} -->

0crat commented:
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

<!-- https://www.0crat.com/footprint/C3NDPUA8L/2391941086, version: 0.30.9, hash: ${buildNumber} -->

0crat commented:
The job [#1656](https://github.com/zerocracy/farm/issues/1656) is now out of scope

<!-- https://www.0crat.com/footprint/C3NDPUA8L/2391944178, version: 0.30.9, hash: ${buildNumber} -->

0crat commented:
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

<!-- https://www.0crat.com/footprint/C3NDPUA8L/2391947321, version: 0.30.9, hash: ${buildNumber} -->

ypshenychka commented:
@krzyk According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

krzyk commented:
@ypshenychka I confirm

ypshenychka commented:
@krzyk thank you

ypshenychka commented:
@0crat quality acceptable

0crat commented:
Order was finished, quality is "acceptable": +20 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

<!-- https://www.0crat.com/footprint/C3NDPUA8L/2401006409, version: 0.30.9, hash: ${buildNumber} -->

0crat commented:
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

<!-- https://www.0crat.com/footprint/C3NDPUA8L/2401006417, version: 0.30.9, hash: ${buildNumber} -->
